flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -489,18 +489,15 @@ public PendingCheckpoint triggerCheckpoint(
 				}
 
 				// make sure the minimum interval between checkpoints has passed
-				final long earliestNext = lastCheckpointCompletionNanos + minPauseBetweenCheckpointsNanos;
-				final long durationTillNextMillis = (earliestNext - System.nanoTime()) / 1_000_000;
+				final long durationTillNextMillis = getDurationTillNextMillis();
```

Comment:
This is good abstraction. Actually we could further abstract the whole logic of check against max concurrent checkpoint attempt and minimum checkpoint interval (L623-645 duplicates L482-503)